From d4b2f923af9dedc136f408f88767f3f85141a895 Mon Sep 17 00:00:00 2001
From: Dmytro Laktyushkin <Dmytro.Laktyushkin@amd.com>
Date: Tue, 30 Aug 2016 18:01:31 -0400
Subject: [PATCH 0951/1722] drm/amd/dal: apply default colours before
 adjustments

Workaround for input scs not being applied for underlay in csc
adjustment code

Change-Id: I6882f66dc8467bc71a2f7df2962ab4e98d6f31bb
Signed-off-by: Dmytro Laktyushkin <Dmytro.Laktyushkin@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c | 11 +++--------
 drivers/gpu/drm/amd/dal/dc/inc/hw/opp.h                 |  3 ++-
 2 files changed, 5 insertions(+), 9 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c
index afbd15d..3e8bea4 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c
@@ -1502,10 +1502,7 @@ static void set_default_colors(struct pipe_ctx *pipe_ctx)
 		pipe_ctx->stream->public.timing.display_color_depth;
 
 	/* Lb color depth */
-	default_adjust.lb_color_depth = LB_PIXEL_DEPTH_30BPP;
-	/*dal_hw_sequencer_translate_to_lb_color_depth(
-			build_params->
-			line_buffer_params[path_id][plane_id].depth);*/
+	default_adjust.lb_color_depth = pipe_ctx->scl_data.lb_bpp;
 
 	pipe_ctx->opp->funcs->opp_set_csc_default(
 					pipe_ctx->opp, &default_adjust);
@@ -1564,6 +1561,7 @@ static void set_plane_config(
 
 	dc->hwss.enable_fe_clock(ctx, pipe_ctx->pipe_idx, true);
 
+	set_default_colors(pipe_ctx);
 	if (pipe_ctx->stream->public.csc_color_matrix.enable_adjustment
 			== true) {
 		tbl_entry.color_space =
@@ -1575,8 +1573,6 @@ static void set_plane_config(
 
 		pipe_ctx->opp->funcs->opp_set_csc_adjustment
 				(pipe_ctx->opp, &tbl_entry);
-	} else {
-		set_default_colors(pipe_ctx);
 	}
 
 	if (pipe_ctx->stream->public.gamut_remap_matrix.enable_remap == true) {
@@ -1927,6 +1923,7 @@ static void dce110_program_front_end_for_pipe(struct core_dc *dc,
 
 	dc->hwss.enable_fe_clock(ctx, pipe_ctx->pipe_idx, true);
 
+	set_default_colors(pipe_ctx);
 	if (pipe_ctx->stream->public.csc_color_matrix.enable_adjustment
 			== true) {
 		tbl_entry.color_space =
@@ -1938,8 +1935,6 @@ static void dce110_program_front_end_for_pipe(struct core_dc *dc,
 
 		pipe_ctx->opp->funcs->opp_set_csc_adjustment
 				(pipe_ctx->opp, &tbl_entry);
-	} else {
-		set_default_colors(pipe_ctx);
 	}
 
 	if (pipe_ctx->stream->public.gamut_remap_matrix.enable_remap == true) {
diff --git a/drivers/gpu/drm/amd/dal/dc/inc/hw/opp.h b/drivers/gpu/drm/amd/dal/dc/inc/hw/opp.h
index 0bcaa0c..ff554ae 100644
--- a/drivers/gpu/drm/amd/dal/dc/inc/hw/opp.h
+++ b/drivers/gpu/drm/amd/dal/dc/inc/hw/opp.h
@@ -27,6 +27,7 @@
 #define __DAL_OPP_H__
 
 #include "hw_shared.h"
+#include "transform.h"
 
 struct fixed31_32;
 struct gamma_parameters;
@@ -220,7 +221,7 @@ enum graphics_csc_adjust_type {
 };
 
 struct default_adjustment {
-	uint32_t lb_color_depth;
+	enum lb_pixel_depth lb_color_depth;
 	enum dc_color_space out_color_space;
 	enum dc_color_space in_color_space;
 	enum dc_color_depth color_depth;
-- 
2.7.4

